tensorflow-opgen/src/TensorFlow/OpGen.hs

```diff
@@ -170,232 +128,206 @@ imports = stack [
     , "import TensorFlow.Output (ResourceHandle)"
     , "import TensorFlow.Tensor"
     , "import TensorFlow.Types"
-      ]
+    ]
+
+renderHaskellName, renderTFName, renderQuotedTFName :: Name -> Doc
+renderHaskellName = strictText . unHaskellName . haskellName
+renderTFName = strictText . unTFName . tfName
+renderQuotedTFName = dquotes . renderTFName
 
-renderDef :: OpDef -> Doc
-renderDef d =
-  stack [
-      haddocks
+
+-- | Generate the source code for a singel op.
+-- For example:
+--
+-- -- | {haddock comment}
+-- foo :: {type sig}
+-- foo attr1 attr2 input1 input2 | eqLengthGuard [...] = {function body}
+renderOp :: ParsedOp -> Doc
+renderOp d = stack $
+    [ haddocks
     , n <+> "::" <+> hang 0 (typeSig d)
-    , n <+> hang 0 args <+> "|" <+> funcGuard <+> "=" </>  -- args are indented
-            -- the body needs to be indented wrt the name
-            indent indentation functionBody
-    , extras  -- just for debug
-    ]
+    , n <+> hang 0 args <+> "|" <+> funcGuard (inferredListSizeAttrs d)
+                <+> "=" </>  -- args are indented
+                    -- the body needs to be indented wrt the name
+                    indent indentation (functionBody d)
+    ] ++ whereClause (inferredListSizeAttrs d)
   where
-    n = strictText $ fixOpName (d ^. name)
-    args = sep $ [hsName | (_, hsName) <- mandatoryAttrs] ++ tensorArgs
-    tensorArgs = [strictText $ lowCase (a ^. name) | a <- d ^. inputArg]
-    fixOpName = lowCase
-    funcGuard = "eqLengthGuard" <+> brackets (commasep entries)
+    n = renderHaskellName $ parsedOpName d
+    args = sep $ map (renderHaskellName . attrName) (explicitInputAttrs d)
+                ++ map (renderHaskellName . parsedArgName) (parsedInputs d)
```

Comment:
Eh, I think it's a little too noisy for my taste due to the "++" mixing with the other operators.  But I factored out the `map renderHaskellName` since I agree it's a little more readable.